### DISCUSSION TITLE
Unable to clear fields using the local API

### DISCUSSION THREAD

**Discussion by allannnc** [payloadcms/payload#3102] [Roadmap]
### Link to reproduction

?

### To Reproduce

**Passing `undefined`**
1. Attempt to clear a field like so:
![image](https://github.com/payloadcms/payload/assets/103010201/2f7b79f8-05af-4130-a1f3-bb320750a31c)
2. The `code` remains the same within the DB (here I'm assuming Payload doesn't know that the `data.code` key was passed intentionally)

<hr />

**Passing `null`**
1. Attempt to clear a field like so:
![image](https://github.com/payloadcms/payload/assets/103010201/6ebdc9c5-a754-48bb-8c4b-7376bcc86a9e)
2. `null` is not allowed by the TS definitions
<hr />
The current workaround is to pass an "empty" but not properly cleared value

![image](https://github.com/payloadcms/payload/assets/103010201/b384f295-e55e-4da6-86b3-f5c28ad22a9a)

**Suggestions**
I think usage of the `in` operator or `hasOwnProperty` could help here to allow passing `undefined`... BUT this would be a breaking change destroying many peoples' data. So perhaps `payload.update`'s `data` arg needs to extend the `generated-types` and allow a magic value to clear the field?

Such as
```ts
await req.payload.update({
  collection: 'coll',
  id: someId,
  data: {
    fieldToClear: 'DELETE_THIS_FIELD'
  }
});
```

### Describe the Bug

As far as I know, there is no way to properly clear or delete a field using the local API

### Payload Version

1.10.1

**Comment by jmikrut** [comment]
Hey @allannnc — your workaround of passing an empty string to the `code` field is the appropriate way to clear the field's data as of now, but we should build a way to "unset" fields via passing `undefined`.

This is a new feature and we will add it to our roadmap for sure. 

Thanks for bringing this up!

**Comment by anthonymzupancic** [comment]
Is there any movement on this or a solution that's been put in place for it? Trying to figure this out for a project.
